api/app/events/api.py

```diff
@@ -96,13 +99,86 @@ def get(self):
         return returnEvents, 200
 
 
+class EventAPI(EventMixin, restful.Resource):
+    @auth_required
+    def post(self):
+        args = self.req_parser.parse_args()
+
+        user_id = g.current_user["id"]
+        current_user = user_repository.get_by_id(user_id)
+        if not current_user.is_admin:
+            return FORBIDDEN
+
+        name = args['name']
+        description = args['description']
+        start_date = datetime.strptime(
+            (args['start_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        end_date = datetime.strptime(
+            (args['end_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+        event = Event(name=name,
+                      description=description,
+                      start_date=start_date,
+                      end_date=end_date)
+
+        db.session.add(event)
+        db.session.commit()
+
+        try:
+
+            admin_event_role = EventRole('admin', user_id, event.id)
+            db.session.add(admin_event_role)
+            db.session.commit()
+        except Exception as e:
+            LOGGER.error(
+                'Failed to add event role for user id {} due to: {}'.format(user_id, e))
+            return errors.ADD_EVENT_ROLE_FAILED
+
+        return event_info(user_id, event), 201
+
+    @auth_required
+    def put(self):
+        args = self.req_parser.parse_args()
+        LOGGER.debug(args)
+        event = db.session.query(Event).filter(
+            Event.id == args['id']).first()
+        if not event:
+            return EVENT_NOT_FOUND
+
+        user_id = g.current_user["id"]
+        event_id = event.id
+        current_user = user_repository.get_by_id(user_id)
+        if not current_user.is_event_admin(event_id):
+            return FORBIDDEN
+
+        name = args['name']
+        description = args['description']
+        start_date = datetime.strptime(
+            (args['start_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+        end_date = datetime.strptime(
+            (args['end_date']), '%Y-%m-%dT%H:%M:%S.%fZ')
+
+        event = db.session.query(Event).filter(
+            Event.id == event_id).first()
+
+        event.name = name
+        event.description = description
+        event.start_date = start_date
+        event.end_date = end_date
```

Comment:
Can we have an update() method on the Event model with these fields that we're updating? Again just to move this logic out of the API.

Reply:
I will add an update method.